code/game/machinery/cryopod.dm

```diff
@@ -241,26 +243,42 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 	container_resist(user)
 
 /obj/machinery/cryopod/process()
-	if(!occupant)
+	if(!occupant || ghost_offering)
 		return
 
 	var/mob/living/mob_occupant = occupant
 	if(mob_occupant)
-		// Eject dead people
-		if(mob_occupant.stat == DEAD)
+		// Eject people that are incapacitated
+		if(mob_occupant.stat)
 			open_machine()
 
+		//The grace period isn't up yet
 		if(!(world.time > despawn_world_time))
 			return
 
-		if(!mob_occupant.client && mob_occupant.stat < 2) //Occupant is living and has no client.
-			if(!control_computer_weakref)
-				find_control_computer(urgent = TRUE)//better hope you found it this time
+		//Offer special roles to ghosts and pause processing while we do
+		var/datum/antagonist/A = mob_occupant.mind.has_antag_datum(/datum/antagonist)
+		if(A)
```

Comment:
This needs to differentiate between soft antagonists like ashwalkers/survivalists (who can just leave and be instantly despawned), conversion/less important antagonists (cult, etc.) who can leave and be offered but despawn if nobody takes them, and important antagonists (traitors, etc.) who need their antag datum to be kept in a round.